src/storage/src/source/source_reader_pipeline.rs

```diff
@@ -1114,17 +1102,10 @@ where
         let mut ticker = tokio::time::interval(timestamp_interval);
         ticker.set_missed_tick_behavior(tokio::time::MissedTickBehavior::Skip);
 
-        tokio::pin!(shutdown_rx);
         let mut input_frontier = Antichain::from_elem(Timestamp::default());
         loop {
             // AsyncInputHandle::next is cancel safe
             tokio::select! {
-                biased;
-                // Make sure we don't accidentally mint new updates when this source has
-                // been dropped. This way, we also make sure to not react to spurious
-                // frontier advancements to `[]` that happen when the input source operator
-                // is shutting down.
```

Comment:
we ensure this by drop being implemented normally with the shutdown button right?

Reply:
Yes, the whole future is dropped when the button is dropped so no more work will happen